Sources/String/Utility.swift

```diff
@@ -142,3 +142,46 @@ extension String {
         return self.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
     }
 }
+
+// MARK: - Slicing using characters
+
+extension String {
+
+    /// Returns a sliced string of given string from `start` character to `end` character.
+    ///
+    /// If the lower bound is greater than the upper bound, it returns the given string.
+    /// If the given string doesn't contain given characters, it also returns the given string.
+    ///
+    /// - Parameters:
+    ///   - start:    The character value being the lower bound of the given string.
+    ///   - end:      The character value being the upper bound of the given string.
+    ///   - inversed: A boolean value that indicates whether the sliced string is inversed. The Default value is `false`.
+    ///
+    /// - Returns: A sliced string of given string.
+    public func sliced(from start: Character, to end: Character, inversed: Bool = false) -> String {
+         guard let startCharacterIndex = firstIndex(of: start),
+           let endCharacterIndex = firstIndex(of: end),
+           startCharacterIndex <= endCharacterIndex
+           else { return self }
+         let slicedString = String(self[startCharacterIndex ... endCharacterIndex])
+         return !inversed ? slicedString : String(slicedString.reversed())
```

Comment:
[언어 공식 문서](https://docs.swift.org/swift-book/LanguageGuide/BasicOperators.html)의 'Ternary Conditional Operator'에 나와 있는 다음의 문단을 미루어 보았을 때 저 실행문에서 true일 때의 값과 false일 때의 값을 모두 평가하지 않는 것으로 보입니다.

> The ternary conditional operator is a special operator with three parts, which takes the form question ? answer1 : answer2. It’s a shortcut for evaluating one of two expressions based on whether question is true or false. If question is true, it evaluates answer1 and returns its value; otherwise, it evaluates answer2 and returns its value.

```swift
func foo1() -> Int { true ? 0 : 1 }
func foo2() -> Int { if true { return 0 } else { return 1 } }
```

위의 코드에 대해 SIL을 뽑아봤고, 두 함수에 대해 중간 코드가 완벽하게 같은 것을 확인할 수 있었습니다.

```text
cond_br %4, bb1, bb2  
```

덧붙여서 모든 함수의 끝에서 위의 문장을 확인할 수 있는데, `%4`(Int1 변수)가 1이면 `bb1` 블록을, 0이면 `bb2` 블록을 평가한다는 의미입니다.

그래서 성능 관점에서 저 문장을 본다면 저렇게 작성해도 이상은 없을 것 같습니다.

Reply:
죄송합니다, 제가 제대로 설명을 못했네요. slicedString을 굳이 String으로 변환을 하고, 뒤에서 또 변환을 하는 부분을 말씀드린 것이였습니다. substring 상태로도 reverse 연산이 가능하기 때문에 String변환을 줄이는 것을 의도 했습니다. 제가 의도한 코드는 다음과 같습니다.

```swift
let sliced = self.[startCharacterIndex...endCharacterIndex]

if invesed {
    return String(sliced)
} else {
    return String(sliced.reversed())
}
```

sliced와 sliced.reversed()의 타입이 달라서, 삼항 연산자를 적용하면 에러가 납니다.

Reply:
앗 이해했습니다. 제시해주신 것이 더 좋겠네요. 수정하겠습니다 🙇‍♂️

Reply:
```swift
String(!inversed ? sliced : sliced.reversed()) // Error! 
!inversed ? String(sliced) : String(sliced.reversed()) // OK
```
이게 더 낫겠네요 ㅎㅎ 괜히 두번 일 하게 해드리네요 ㅜ